libraries/AP_Telemetry/MQTT_Mavlink.cpp

```diff
@@ -0,0 +1,275 @@
+    ret = 0;
+    AP_Telemetry_MQTT* tele_mqtt = AP_Telemetry_MQTT::get_telemetry_mqtt();
+    char topic[MAX_TOPIC];
+
+    printf("received mqtt from Pc %s \n", cmd);
+    if (strncmp(cmd, "arm", 3) == 0) {
+        sprintf(topic, "$ardupilot/copter/quad/ack/%04d", mavlink_system.sysid);
+        //arm コマンド発行
+        memset(msg, 0, sizeof(mavlink_message_t));
+        msg->msgid = MAVLINK_MSG_ID_COMMAND_LONG;
+        mavlink_msg_command_long_pack_chan(
+            0,0,0,
+            msg,
+            0,0,MAV_CMD_COMPONENT_ARM_DISARM,0,
+            1.0,0.0,0.0,0.0,0.0,0.0,0.0);
+        tele_mqtt->send_message("Arm command received", topic);
+        ret = 1;
+    } else if (strncmp(cmd, "disarm", 6) == 0) {
+        sprintf(topic, "$ardupilot/copter/quad/ack/%04d", mavlink_system.sysid);
+        tele_mqtt->send_message("Disarm command received", topic);
+        // disarm コマンド発行
+        memset(msg, 0, sizeof(mavlink_message_t));
+        msg->msgid = MAVLINK_MSG_ID_COMMAND_LONG;
+        mavlink_msg_command_long_pack_chan(
+            0,0,0,
+            msg,
+            0,0,MAV_CMD_COMPONENT_ARM_DISARM,0,
+            0.0,0.0,0.0,0.0,0.0,0.0,0.0);
+        ret = 1;
+    } else if (strncmp(cmd, "takeoff", 7) == 0) {
+        float takeoff_alt = 20;// param7
+        takeoff_alt = atof(&cmd[8]);
+        char payload[MAX_PAYLOAD];
+        sprintf(topic, "$ardupilot/copter/quad/ack/%04d", mavlink_system.sysid);
+        sprintf(payload, "Took off to %f", takeoff_alt);
+        tele_mqtt->send_message(payload, topic);
+        if (strlen(cmd) >= 9) {
+            takeoff_alt = atof(&cmd[8]);
+        }
+        float hnbpa = 1.0; // param 3 horizontal navigation by pilot acceptable
+        memset(msg, 0, sizeof(mavlink_message_t));
+        msg->msgid = MAVLINK_MSG_ID_COMMAND_LONG;
+        mavlink_msg_command_long_pack_chan(
+            0,0,0,
+            msg,
+            0,0,MAV_CMD_NAV_TAKEOFF,0,
+            0.0,0.0,hnbpa,0.0,0.0,0.0,takeoff_alt);
+        ret = 1;
+    } else if (strncmp(cmd, "mode land", 9) == 0) {
+        //mode land
+        char buf[30] = {0};
+        float hnbpa = 1.0; // param 3 horizontal navigation by pilot acceptable
+        msg->msgid = MAVLINK_MSG_ID_COMMAND_LONG;
+        mavlink_msg_command_long_pack_chan(
+            0,0,0,
+            msg,
+            0,0,MAV_CMD_NAV_LAND,0,
+            0.0,0.0,hnbpa,0.0,0.0,0.0,0.0);
+        msg->msgid = MAVLINK_MSG_ID_SET_MODE;
+        msg->len = 6;
+        _mav_put_uint32_t(buf, 0,9);
+        _mav_put_uint8_t(buf, 4,1);
+        _mav_put_uint8_t(buf, 5,1);
+        memcpy(_MAV_PAYLOAD_NON_CONST(msg), buf, 30);// 30 is about
+        memset(msg, 0, sizeof(mavlink_message_t));
+        sprintf(topic, "$ardupilot/copter/quad/ack/%04d", mavlink_system.sysid);
+        tele_mqtt->send_message("Landing receive", topic);
+        ret = 1;
+    } else if ((strncmp(cmd, "mode alt_hold", 13) == 0) || (strncmp(cmd, "mode althold", 12) == 0)) {
```

Comment:
I write the number of characters directly. If you define a character string, I think you can use strlen.